### ISSUE TITLE
Document core design philosophies

### ISSUE THREAD

**Issue by MythicManiac** [WolvenKit/gpm#7]
It could be a good idea to document our core design philosophies to guide our design choices. Something to start off with listed below.

- Put developers first
- The CLI should be an effective tool for mod package development
- The CLI should be an effective tool for mod package management
- Mod managers should be able to rely on the CLI for their functionality
- The GPM project should completely take care of
  - Dependency management
  - Package building
  - Package publishing
  - Package distribution
  - Package downloading
- The GPM project should facilitate package installation
  - Package installation should be kept versatile and extendable

**Comment by Aelto** [comment]
You're right, it is time we create the README files for this and eventually the wiki.
